src/XMLimport.cpp

```diff
@@ -149,7 +152,37 @@ bool XMLimport::importPackage(QIODevice* device, QString packName, int moduleFla
         readNext();
 
         if (isStartElement()) {
-            if (name() == "MudletPackage") {
+            if (name() == QStringLiteral("MudletPackage")) {
+                QString versionString;
+                if (attributes().hasAttribute(QStringLiteral("version"))) {
+                    versionString = attributes().value(QStringLiteral("version")).toString();
+                    if (!versionString.isEmpty()) {
+                        bool isOk = false;
+                        float versionNumber = versionString.toFloat(&isOk);
+                        if (isOk) {
+                            mVersionMajor = qFloor(versionNumber);
+                            mVersionMinor = qRound(1000.0 * versionNumber) - (1000 * mVersionMajor);
+                        }
+                        if (pVersionString) {
+                            *pVersionString = versionString;
+                        }
+                    }
+                }
+
+                if (mVersionMajor > 1
+                    /*||(mVersionMajor==1&&mVersionMinor)*/) {
+                    // Minor check is not currently relevant, just abort on 2.000f or more
+
+                    QString moanMsg = tr("[ ALERT ] - Sorry, the file being read is marked as being version %1 which is\n"
```

Comment:
> "Sorry, this settings file:\n%s\ncomes from a newer version of Mudlet and can't be understood - please upgrade your Mudlet"

Humm, is "settings" file the right term - it can be any of:
+ Mudlet profile save file
+ a "package" containing one or more of "items" of one type
+ the Mudlet core/guts at the root of a "module".

I will need to check to see how to get the PathFileName to report as this method is working with a `QIODevice*` and I will have to check there is always a file on the other end of that - IIRC sometimes we might be use a buffer in memory.

I think we need to report the version found simply so when someone asks we can say "yes, your version of Mudlet __is__ too old to read that type of file; you need version 5.1 or later and ideally you will find 6.0 which we released last month and that will really frob that file."